### PR TITLE
Fe/feat/graph ql

### PR DESCRIPTION
**Pull request by andreasorensen**
# Description

- implemented first useQuery for mushroomMatches 
 - rendered mushroomCard matches
 
 Note: ** need to figure out the rendering issue

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
    - ** does break on some photos -- is it because there is no match found on the API? 
    - ** giving "too many renders", 500 error, and ApolloError for undefined method "first for nil-NilClass 
   
- [ ] This change requires a documentation update
- [ ] Testing
- [ ] Refactoring

# How Has This Been Tested?

No yet

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,12 +4,23 @@ import './index.css';
...
+  cache: new InMemoryCache(),
+});
+
 const root = ReactDOM.createRoot(document.getElementById('root'));
 root.render(
-  <BrowserRouter>
-    <React.StrictMode>
-      <App />
-    </React.StrictMode>
+  <ApolloProvider client={client}>
```
Comment (jmartin777) [inline src/index.js:16]: really great work here connecting this in the root with Apollo

File: src/index.js [hunk 2]
```diff
@@ -4,12 +4,23 @@ import './index.css';
 import App from './Components/App/App';
 import { BrowserRouter } from 'react-router-dom';
 
+import { ApolloClient, InMemoryCache, ApolloProvider } from "@apollo/client";
```
Comment (jmartin777) [inline src/index.js:7]: that new Tech

**Review by jmartin777 — APPROVED** [review]
Really great work here Andrea getting this going and connected!

File: src/Components/App/App.js [hunk 3]
```diff
@@ -11,21 +11,41 @@ import MushroomCard from "../MushroomCard/MushroomCard";
...
 
 function App() {
   const [showError, setShowError] = useState(false);
   const [error, setError] = useState(null);
   const [userImage, setUserImage] = useState(null);
 
+  const { loading, error: queryError, data } = useQuery(GET_MUSHROOM_MATCHES, {
+    variables: { image: userImage },
+    skip: !userImage,
+  });
```
Comment (Elise-Jones) [inline src/Components/App/App.js:26]: So happy this useQuery is connecting for the most part.

**Review by Elise-Jones — APPROVED** [review]

**Pull request merged**